python_modules/dagster/dagster/_core/definitions/assets.py

```diff
@@ -858,6 +871,27 @@ def check_specs(self) -> Iterable[AssetCheckSpec]:
         """
         return self._check_specs_by_output_name.values()
 
+    @property
+    def asset_check_handles(self) -> AbstractSet[AssetCheckHandle]:
+        """Returns the selected asset checks associated by this AssetsDefinition.
+
+        Returns:
+            AbstractSet[Tuple[AssetKey, str]]: The selected asset checks. An asset check is
+                identified by the asset key and the name of the check.
+        """
+        return self._selected_asset_check_handles
+
+    @public
+    @property
+    def check_handles(self) -> AbstractSet[Tuple[AssetKey, str]]:
```

Comment:
I think we should expose `AssetCheckHandle` publicly rather than tuples. Also, I don't think we need both a `check_handles` and an `asset_check_handles` property.

Reply:
Yeah I pushed back on a type since I didn't want to intoroduce a new concept, handles to the public API. However a public property called `check_handles` does the same thing.

How about we do `selected_check_specs` and return a list of `AssetCheckSpec`?

Reply:
I don't feel strongly about the name at all, but I feel pretty strongly that we should expose a type for this.  By the fact that a check is globally identifiable by its (asset key, check name), the concept is inherently present in the ontology.  Hiding it will just require a set of contortions.

Reply:
Do we have any other analogous concepts? Searching for a name here.

It is our first instance of an asset-scoped identifier, although `PartitionKey` might count.

What about `AssetCheckKey`? This stays consistent with the naming that a "Key" is a globally unique identifier for a spec in this system.

Reply:
I like `AssetCheckKey`.

We use `AssetKeyPartitionKey` internally, which would correspond to something like `AssetKeyCheckName`, though a bit of a mouthful.

Reply:
Why don't we say `AssetPartitionKey`?

Reply:
The previous concern with AssetCheckKey is that it seems like it could overlap with AssetKey. (side note, you currently can get duplicate op name errors since `op_name = {asset_key}_{check_name}`).

I'm happy with AssetCheckKey but just surfacing before we commit

Reply:
> side note, you currently can get duplicate op name errors since op_name = {asset_key}_{check_name}

if key + check name are guaranteed to be unique, don't we have a uniqueness guarantee?

Reply:
key + check name are guaranteed unique across checks. We concat them to create Op names which is where you can hit a conflict

Reply:
I am still confused. Asset keys should be globally unique within a deployment. Can you provide a concrete example of a collision?

Reply:
- asset_key = "a", check_name = "b_c"
- asset_key = "a_b", check_name = "c"

Reply:
We should mangle the at least a little bit to make that exceeding unlikely.

Reply:
I'll pick this conversation back up in a thread